### PR TITLE
allow all --catchup command line options to operate on the same in-me…

### PR DESCRIPTION
**Pull request by vogel**
…mory sqlite database

Signed-off-by: Rafał Malinowski <rafal.przemyslaw.malinowski@gmail.com>

### CODE REVIEW

File: src/main/main.cpp [hunk 1]
```diff
@@ -262,12 +262,9 @@ checkInitialized(Application::pointer app)
 }
 
 static int
-catchup(Config const& cfg, uint32_t to, uint32_t count,
+catchup(Application::pointer app, uint32_t to, uint32_t count,
         Json::Value& catchupInfo)
 {
-    VirtualClock clock(VirtualClock::REAL_TIME);
```
Comment (MonsieurNicolas) [inline src/main/main.cpp]: why not do `VirtualClock& clock = app->getClock()`, this avoids having all those `app->getClock()` calls

File: src/main/main.cpp [hunk 2]
```diff
@@ -856,22 +850,31 @@ main(int argc, char* const* argv)
...
-                result = catchupTo(cfg, catchupToTarget, catchupInfo);
-            if (!catchupInfo.isNull())
-                writeCatchupInfo(catchupInfo, outputFile);
+            if ((result == 0) && (doCatchupAt || doCatchupComplete ||
+                                  doCatchupRecent || doCatchupTo))
+            {
+                Json::Value catchupInfo;
+                VirtualClock clock(VirtualClock::REAL_TIME);
+                auto app = Application::create(clock, cfg, false);
+                if ((result == 0) && doCatchupAt)
```
Comment (MonsieurNicolas) [inline src/main/main.cpp]: `result == 0` is redundant on the first line

**Comment by vogel** [comment]
Updated

**Comment by MonsieurNicolas** [comment]
r+ 8a4fafc384f138172b89f120dfc8cc0702ff811e

**Comment by MonsieurNicolas** [comment]
@latobarita: retry

**Pull request merged**